app/frontend/src/components/autocomplete/autocomplete.js

```diff
@@ -1,37 +1,46 @@
-import 'es6-promise/auto';
-import view from './autocomplete.view';
+import accessibleAutocomplete from 'accessible-autocomplete';
+import 'accessible-autocomplete/dist/accessible-autocomplete.min.css';
+import './autocomplete.scss';
 
-export const renderAutocomplete = (widgetParams) => {
-  view.create(widgetParams.container, widgetParams.input, widgetParams.key);
+const AUTOCOMPLETE_THRESHOLD = 3;
 
-  widgetParams.input.addEventListener('input', () => {
-    if (isActive(widgetParams.threshold, widgetParams.input.value.length)) {
-      autocomplete.showOptions(widgetParams.getOptions, widgetParams.container, widgetParams.input, widgetParams.key);
-    } else {
-      autocomplete.clearOptions(widgetParams.container, widgetParams.input);
-    }
-  });
-};
-
-export const removeUnwantedCharacters = (searchTerm) => searchTerm.replace('.', ' ');
+const highlightRefinement = (text, refinement) => {
+  const index = text.toLowerCase().indexOf(refinement.toLowerCase());
 
-export const showOptions = (getOptions, container, input, key) => {
-  const searchTerm = removeUnwantedCharacters(input.value);
-  return getOptions(searchTerm).then((options) => view.show(options, container, input, key));
-};
+  if (index >= 0) {
+    /* eslint-disable */
+    return `${text.substring(0, index)}<span class='accessible-autocomplete__suggestion-highlight'>${text.substring(index, index + refinement.length)}</span>${text.substring(index + refinement.length, text.length)}`;
+    /* eslint-enable */
+  }
 
-export const clearOptions = (container, input) => {
-  view.hide(container, input);
+  return text;
 };
 
-export const isActive = (threshold, inputLength) => inputLength >= threshold;
-
-const autocomplete = {
-  view,
-  renderAutocomplete,
-  removeUnwantedCharacters,
-  showOptions,
-  clearOptions,
+const autocomplete = (fieldIds, source) => {
+  fieldIds.forEach((elementId) => {
+    const formInput = document.getElementById(elementId);
+
+    if (formInput) {
+      let currentInputValue = formInput.value;
+      formInput.parentNode.removeChild(formInput);
+
+      accessibleAutocomplete({
+        element: document.querySelector('#accessible-autocomplete'),
+        id: elementId,
+        name: 'location',
+        defaultValue: currentInputValue,
+        source: (query, populateResults) => {
+          currentInputValue = query;
+          return source({ query, populateResults });
```

Comment:
is `populateResults` something defined by accessible autocomplete?

Reply:
yes it an argument passed into this callback

Reply:
although you could call it whatever you want, this is name in their docs, so ill leave it as is in case someone ever cross references the two